crates/sui-core/src/authority/authority_store.rs

```diff
@@ -427,25 +428,26 @@ impl AuthorityStore {
                             epoch_store.get_shared_locks(digest)?.into_iter().collect(),
                         )
                     })?;
-                    match shared_locks.get(id) {
-                        Some(version) => {
-                            if let Some(obj) = self.get_object_by_key(id, *version)? {
-                                result.push(obj);
-                            } else {
-                                // When this happens, other transactions that use smaller versions of
-                                // this shared object haven't finished execution.
-                                errors.push(SuiError::SharedObjectPriorVersionsPendingExecution {
-                                    object_id: *id,
-                                    version_not_ready: *version,
-                                });
-                            }
-                            continue;
-                        }
-                        None => {
-                            errors.push(SuiError::SharedObjectLockNotSetError);
-                            continue;
-                        }
+                    // If we can't find the locked version, it means
+                    // 1. either we have a bug that skips shared object verison assignment
+                    // 2. or we have some DB corruption
+                    let version = shared_locks.get(id).unwrap_or_else(|| {
+                        panic!(
+                        "Shared object locks should have been set. tx_digset: {:?}, obj id: {:?}",
+                        digest, id
+                    )
+                    });
+                    if let Some(obj) = self.get_object_by_key(id, *version)? {
+                        result.push(obj);
+                    } else {
+                        // When this happens, other transactions that use smaller versions of
+                        // this shared object haven't finished execution.
+                        errors.push(SuiError::SharedObjectPriorVersionsPendingExecution {
```

Comment:
After TransactionManager, this `SharedObjectPriorVersionsPendingExecution` error here should panic as well.

Reply:
you are right, I'm removing it as well

Reply:
actually I'm going to do that in a separate PR

Reply:
https://github.com/MystenLabs/sui/pull/8507